cpp/src/arrow/csv/reader.h

```diff
@@ -25,6 +25,7 @@
 #include "arrow/result.h"
 #include "arrow/type.h"
 #include "arrow/type_fwd.h"
+#include "arrow/util/async_generator.h"
```

Comment:
```suggestion
```

I don't think this is needed?  We have been mostly trying to treat `async_generator.h` as an internal header as it includes quite a few different things and can bloat header sizes.

Can you put this in `reader.cc` if needed?